testy/AktivitaTagyTest.php

```diff
@@ -51,9 +50,18 @@ function testKopiePriInstanciaci($idAktivity, $_, $tagy) {
     $a = Aktivita::zId($idAktivity);
     $a->nastavTagy($tagy);
     $b = $a->instanciuj();
-    $this->assertEquals($tagy, $b->tagy(),
+    $this->assertEquals(serazene($tagy), serazene($b->tagy()),
       "Tagy se musí propsat i do nově vytvořené instance"
     );
   }
 
 }
+
+/**
+ * Vrátí seřazenou kopii pole bez modifikace původního pole.
+ */
+function serazene($pole) {
+  $serazene = $pole;
```

Comment:
Úplná prkotina, jenom pro tvou informaci - když v PHP strčíš pole do funkce a pak na to pole sáhneš (třeba funkcí sort), tak si PHP vytoří kopii původního pole a změnu provede až na ní. Originál zůstane netknutý. To by muselo být originální pole posláno do funkce jako reference.
Mám ale pocit, že nějaká stará verze PHP to tak ještě neměla, něco jako 5.5, takže si to možná pamatuješ z toho.

Reply:
V pohode, ja vim jak funguji pole v php, jenom nechci modifikovat parametr funkce (zvlast kdyz pak tu promennou vracim).

Reply:
To mě jako programátora z povolání zajímá. Ta funkce má jediný účel (což je super) a je úplně jedno, co s tím vstupním parametrem dělá, protože navenek to nic neovlivní. Tím že výslovně vytvoříš novou kopii toho pole, tak donutíš systém, aby držel v paměti dvě stejné struktury a když začneš kopii řadit, tak systém musí udržovat originál netknutý. Kdyby to byly statisíce položek, tak to může slušně vyžrat RAMku a samotné kopírování taky nějaký čas sežere.
Nesahat na vstupní parametry je dobrý zvyk a zejména v tomhle kontextu je ta kopie pole úplně putna, ale zajímá mě důvod, abych na to myslel v extrémních případech.

Reply:
No, ten duvod je, ze nechci modifikovat parametr funkce, protoze delat to je matouci, a tady to neni performance sensitive. To je cele :)